firebase-firestore/src/main/java/com/google/firebase/firestore/FirestoreRegistrar.java

```diff
@@ -41,13 +44,17 @@
         Component.builder(FirestoreMultiDbComponent.class)
             .add(Dependency.required(FirebaseApp.class))
             .add(Dependency.required(Context.class))
+            .add(Dependency.required(HeartBeatInfo.class))
```

Comment:
These should be optional dependencies, shouldn't they?

Reply:
These should be required? Bcz this assumes we have the firebase dependencies during component initialization. The constructor should have nulllable attributes so that they can be used in the tests. Am I understanding this currently?

cc @vkryachko

Reply:
Logically it seems optional, as it is not critical to the operation of Firestore, however I don't think it matters much as the components in question are part of `firebase-common` so there should not be a case where they would be missing at runtime, this could in theory change in the future though.